walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,93 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
+
+  /**
+   * This function returns a list of potential time ranges for a meeting 
+   * during the day. The request contains the attendees and the duration
+   * of the meeting, and events contains the other events that are 
+   * scheduled already. If an attendee of the meeting request is an
+   * attendee of another event, the times that the other event covers
+   * are no longer possible for the meeting.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    
+    // If the request is over a day, then there are no time slots available
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return possibleTimes;
+    }
+
+    // Start with the whole day available
+    possibleTimes.add(TimeRange.WHOLE_DAY);
+
+    if (request.getAttendees().size() == 0) {
+        return possibleTimes;
+    }
+
+    // Only update times for events that have attendees from the meeting request
+    for (Event event : events) {
+        for (String attendee : request.getAttendees()) {
+            if (event.getAttendees().contains(attendee)) {
+                updatePossibleTimes(possibleTimes, event.getWhen(), request.getDuration());
+            }
+        }
+    }
+
+    return possibleTimes;
+  }
+
+  /** 
+   * Update possible meeting times by excluding time ranges of an event
```

Comment:
One thing that would be useful to do here and on the javadoc of `splitTimeRanges` is explain what is happening with an example. For instance, `If possibleTimes={[12am,1am],[3am, 4am]}, eventTime=[3.30-4.30am], requestDuration=1hr -----> return {[12am,1am]}.